### ISSUE TITLE
Relevant papers for astronomical scheduling

### ISSUE THREAD

**Issue by yoachim** [astropy/astroplan#159]
A scheduling model for astronomy
M. Solara, , , P. Michelonb, J. Avariasc, d, M. Garcesd
http://www.sciencedirect.com/science/article/pii/S2213133716300142
(Scheduling ALMA with a mixed integer linear programming)

An Integer Linear Programming Solution to the Telescope Network Scheduling Problem
Sotiria Lampoudi, Eric Saunders, Jason Eastman
http://arxiv.org/abs/1503.07170

**Comment by eteq** [comment]
See the bibliography at http://constraints.cs.washington.edu/cassowary/  - the *application* of that is quite different, but I think it's actually very mappable onto the scheduling problem.  Equally usefully, the toolkit in question has a python wrapper: https://github.com/enthought/casuarius

**Comment by StuartLittlefair** [comment]
See also http://scheduler.dc3.com/Scheduler2006.pdf

**Comment by wtgee** [comment]
Hi folks. I'm at SPIE in Edinburgh right now and scheduling is a topic that is coming up pretty frequently within the software talks. A lot of people have this wrapped up in their Observatory Control Software packages (of which many people are building separately) but many of them are also independently writing up their own schedulers. I've been trying to turn people onto the discussion here but am not sure if that will just add more noise to this particular group or not.

What I'd really like to do is organize some kind of Working Group or forum where people could be discussing these issues as 80% of the problems are the same across observatories and it is that 80% (with the ability to easily solve the remaining 20%) that I think astroplan is aiming to cover.  

There are a lot of people out there who have spent considerable time thinking about these issues and it would be a shame to not be incorporating them. What are the thoughts here about somehow broadening this discussion to the wider astronomy community (including non-python users) beyond just a posting of a few of the relevant papers?

**Comment by bmorris3** [comment]
> What are the thoughts here about somehow broadening this discussion to the wider astronomy community (including non-python users) beyond just a posting of a few of the relevant papers?

What do you propose? Maybe the best thing would be to get those people who have experience involved directly, so I appreciate that you're trying to do that.

**Comment by wtgee** [comment]
Yes, mostly I am trying to get people involved in some sense. Some of them are not Python though but still have relevant and good ideas concerning best practices, etc. Honestly I don't know much about the subject at this point so don't really know what exists out there, it was just blindingly obvious to me (from a software developer standpoint) that many astronomers are recreating this wheel and they all seem to be doing it independently. 

I'll poke around at SPIE a little more tomorrow and maybe bring the subject up in one of their ad hoc lightning talks, but I wanted to check in with this group first to make sure I wasn't myself just recreating another wheel by starting another discussion.

It seems like a Working Group or something might be good, I'm just not really sure who that is organized through. Any ideas appreciated, but again I don't want to just up the noise.

**Comment by bmorris3** [comment]
@wtgee I think the most helpful input that they could give is to suggest scheduling algorithms. Implementations in Python aren't important, because we can figure out how to code them, but we need guidance towards the best algorithms used to optimize schedules for astronomical problems.

**Comment by yoachim** [comment]
I'm also chillin here at SPIE and can confirm that there's lots of telescope scheduler discussion going on.  

Really, it sounds like we need someone to organize a conference on telescope scheduling...get all the ground and space-based folks in the same room, invite a bunch of the Operations Research folks...[NOT IT].  But it would be pretty cool to have a few days of presentations and a few hack days to put together a framework where people could then drop in their own scheduling algorithms.

**Comment by ebellm** [comment]
Here's another nice paper that includes a fair bit of literature review:

http://adsabs.harvard.edu/abs/2012SPIE.8448E..1LC

My own sense is that it's only useful to think about algorithms once you've decided clearly on the merit function you're trying to optimize--open shutter efficiency? fairness between TAC-allocated programs?  observation utility towards a specific scientific goal?

**Comment by bmorris3** [comment]
See also https://ui.adsabs.harvard.edu/abs/2019BAAS...51g.125B/abstract.
